poetry/vcs/git.py

```diff
@@ -22,7 +22,7 @@
     re.compile(
         r"^(git\+)?"
         r"(?P<protocol>https?|git|ssh|rsync|file)://"
-        r"(?:(?P<user>.+)@)*"
+        r"(?:(?P<user>\w+)@)*"
```

Comment:
```suggestion
        r"(?:(?P<user>[a-zA-Z0-9_.-])@)*"
```

I think we only want standard characters here with the addition of `_`, `.`  and `-`. If you think `\w` is still needed, however I don't think so, adding those three characters is still needed.

Reply:
What ever works is fine for me :) 

I thinks we should use the same pattern for the same groups. For example `resource` is one time `[a-z0-9_.-]*` and one time `[\w.\-]+`. But I'm not totally sure, what is allowed in each group. Any suggestions?

Reply:
I agree we should have consistency between the different regexps. I think for `resource`  `[a-z0-9_.-]+` should be enough.

Reply:
I did some rework to be consistent within the pattern. I also added tests for `parse_url` which uncovered some problems with regexes. Now only the first of the two pattern for matching "protocol://" and for matching "user@" are used for our test data. Do you think we can remove the two other? The less regex we need, the better :smiley: